### ISSUE TITLE
Use pandas' `date_range()`

### ISSUE THREAD

**Issue by Peter9192** [WAM2layers/WAM2layers#29]
There's quite a bit of code for dealing with dates, leap years, etc.

Pandas has a nice utility called [`date_range()`](https://pandas.pydata.org/docs/reference/api/pandas.date_range.html) that can do all of this in one line, automatically taking care of leap years. For example

```py
In [1]: import pandas as pd

In [2]: pd.date_range(start="2002", end="2003", freq="d", inclusive="left")
Out[2]:
DatetimeIndex(['2002-01-01', '2002-01-02', '2002-01-03', '2002-01-04',
               '2002-01-05', '2002-01-06', '2002-01-07', '2002-01-08',
               '2002-01-09', '2002-01-10',
               ...
               '2002-12-22', '2002-12-23', '2002-12-24', '2002-12-25',
               '2002-12-26', '2002-12-27', '2002-12-28', '2002-12-29',
               '2002-12-30', '2002-12-31'],
              dtype='datetime64[ns]', length=365, freq='D')
```

This is more readable and less error-prone. For flexibility, it might be good to pass in start- and end date as "20020101" and perhaps also the frequency (for ERA5 is available hourly). Note that `inclusive="left"` and setting the end to 2003 ensures that daily ranges end at 2002-12-31 and hourly ranges end at 2002-12-31 23:00.

**Comment by ruudvdent** [comment]
I agree this is a good idea, but I think eventually we still want to be in numpy when the bigger calculations start for speed purposes

**Comment by Peter9192** [comment]
> I agree this is a good idea, but I think eventually we still want to be in numpy when the bigger calculations start for speed purposes

Good point. In this case I don't think it interferes too much, but when it becomes relevant, it would be quite straightforward to switch from pandas dataframes/xarray datasets to the underlying numpy arrays.
